src/System/Net/Benchmarks/Common/Shared/SingleStreamConnection.cs

```diff
@@ -3,9 +3,18 @@
 
 namespace System.Net.Benchmarks;
 
-internal abstract class SingleStreamConnection<TStream> : IConnection
```

Comment:
There were no abstract method in this class, so I deleted it.

Reply:
I believe we'd need to revert the changes in this file.

It is abstract because it cannot be used directly. The protected `InnerStream` property is technically playing the role of the "abstract method". There's no actual abstract method bc the class is used as a base class in both server and client, and they have different consuming methods.

We can pull the full single-stream client connection implementation -- but only a part of server connection implementation -- "up". But it would still leave `SingleStreamConnection` abstract.

There's how I see it:

<details>

**1. (move "up" from `Tls` to `Common`)**
```diff
-   ITlsBenchmarkClientConnection
+   IStreamingClientConnection<TOptions>
```

**2. add `IStreamingServerConnection : IConnection` with `AcceptInboundStreamAsync` method pulled up from `ITlsBenchmarkServerConnection`**

**3. (remove `AcceptInboundStreamAsync` from `ITlsBenchmarkServerConnection`)**
```diff
-   ITlsBenchmarkServerConnection : IConnection
+   ITlsBenchmarkServerConnection : IStreamingServerConnection
```

**4. Keep `SingleStreamConnection` abstract**

**5. (move "up" from `Apps` to `Common`)**
```diff
-   SslStreamClientConnection : SingleStreamConnection<SslStream>, ITlsBenchmarkClientConnection
+   SingleStreamConnectionClientConnection<TStream, TOptions> : SingleStreamConnection<TStream>, IStreamingClientConnection<TOptions>
```

**6. add `SingleStreamServerConnection<TStream, TOptions> : SingleStreamConnection<TStream>, IStreamingClientConnection<TOptions>` with `AcceptInboundStreamAsync` method impl pulled up from `SslStreamBenchmarkServerConnection`**

**7. (remove `AcceptInboundStreamAsync` from `SslStreamBenchmarkServerConnection `)**
```diff
-   SslStreamBenchmarkServerConnection : SingleStreamConnection<SslStream>, ITlsBenchmarkServerConnection
+   SslStreamBenchmarkServerConnection : SingleStreamServerConnection<SslStream>, ITlsBenchmarkServerConnection
```

</details>

This definitely can be done as a follow-up (I mean, not in this PR), and it is even still up to discussion. It technically might be more trouble than worth, but it can support the positioning of "NetworkStream" benchmarks alongside Tls benchmarks as a group of "streaming" benchmarks.

Reply:
I'm planning to do this as a follow-up, I don't want to mix refactoring with this PR.